acl/src/main/java/org/apache/rocketmq/acl/plain/RemoteAddressStrategyFactory.java

```diff
@@ -152,6 +192,27 @@ private boolean analysis(String[] strArray, int index) {
             return this.end > 0 ? true : false;
         }
 
+        private boolean ipv6Analysis(String[] strArray, int index) {
+            String value = strArray[index].trim();
+            this.index = index;
+            if ("*".equals(value)) {
+                int min = Integer.parseInt("0", 16);
+                int max = Integer.parseInt("ffff", 16);
+                setValue(min, max);
+            } else if (AclUtils.isMinus(value)) {
+                if (value.indexOf("-") == 0) {
+                    throw new AclException(String.format("RangeRemoteAddressStrategy netaddress examine scope Exception value %s ", value));
+                }
+                String[] valueArray = StringUtils.split(value, "-");
+                this.start = Integer.parseInt(valueArray[0], 16);
+                this.end = Integer.parseInt(valueArray[1], 16);
+                if (!(AclUtils.isIPv6Scope(end) && AclUtils.isIPv6Scope(start) && start <= end)) {
+                    throw new AclException(String.format("RangeRemoteAddressStrategy netaddress examine scope Exception start is %s , end is %s", start, end));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: start (declared at line 207), end (declared at line 208). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 210. Consider formatting this data with &#39;%d&#39; instead.